New Tab Page: Prompt for Brave Together

petemill opened the pull request:
- Button in footer always shows
- Prompt (anchored to button) shows like a tooltip until the user manually closes via button.
- _(Also adds dir="rtl" as attribute in instead of stylesheet so that it's targetable as a css selector. This is done in most chromium webui now.)_

<!-- Add brave-browser issue bellow that this PR will resolve -->
Resolves https://github.com/brave/brave-browser/issues/12776

![image](https://user-images.githubusercontent.com/741836/103858837-e9da7e80-506d-11eb-9281-1f96075ee5fd.png)

![image](https://user-images.githubusercontent.com/741836/103858937-1098b500-506e-11eb-8824-f5d120ddc5c3.png)

### Button Transitions

![button-transitions](https://user-images.githubusercontent.com/741836/103929181-c3e6c580-50d1-11eb-929f-78d431c9f4fd.gif)


## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed).
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

petemill commented on components/brave_new_tab_ui/components/default/footer/togetherTooltip/button.tsx:
```diff
@@ -0,0 +1,59 @@
+
+import styled from 'brave-ui/theme'
+
+const Button = styled('button')`
```
Apparently this is a new design system-able button style that we can make more shared when the design team iterates a bit more

zenparsing commented on components/brave_new_tab_ui/reducers/new_tab_reducer.ts:
```diff
@@ -186,6 +187,14 @@ export const newTabReducer: Reducer<NewTab.State | undefined> = (state: NewTab.S
       break
     }
 
+    case Actions.dismissTogetherPrompt.getType(): {
```
It looks like you are preferring not to add an exported constant to `new_tab_types`. Do you want to eventually remove the need for that file completely?

petemill replied:
Yes just trying to remove 1 layer of boilerplate. I've been using `redux-act` for Brave Today on the NTP via `today_actions.ts` and `reducers/today`, so just continuing that here for anything new.

zenparsing commented on components/brave_new_tab_ui/components/default/footer/togetherTooltip/style.ts:
```diff
@@ -0,0 +1,114 @@
+    left: 0;
+    right: 0;
+    background: linear-gradient(305.95deg, #BF14A2 0%, #F73A1C 98.59%);
+    clip-path: polygon(
```
Neat!

petemill commented on components/brave_new_tab_ui/components/default/footer/togetherTooltip/button.tsx:
```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2020 The Brave Authors. All rights reserved.
```
oops, it's 2021 now

petemill replied:
fixed

petemill commented:
Rebased in order to try to fix CI sync errors. Also added a change to storybook in order to provide an option for right-to-left so that we can easily preview what the browser does on the NTP for certain languages. https://github.com/brave/brave-core/pull/7532/commits/d880de20c1511e89c63bf8c5384fb3807f1a90aa

karenkliu commented:
@petemill The icon in the tooltip prompt should appear on the right side before the title string for right-to-left languages:

![Frame 1](https://user-images.githubusercontent.com/2388823/103933353-17f4a880-50d8-11eb-8612-2cf9e15106c6.png)
